saas/models.py

```diff
@@ -1528,6 +1528,26 @@ def provider(invoiced_items):
             result = get_current_provider()
         return result
 
+    def by_organization(self, organization, transaction_type):
```

Comment:
I think of this as "all transactions of a certain transaction type (account), to or from the specified organization." This was code that appeared in two places, and I needed to add a third, so I unified and refactored it.

The weird bit about your proposal is that the return type of either `by_organization` or `by_account` is not really a `QuerySet`, if the whole expression produces the same filters as the original in this patch. That's because we want a complex Q expression-style thing - for instance, what would this do? `Transaction.objects.by_organization(org).filter(orig_amount > 100000)`?

Or worse, `Transaction.objects.by_organization(org).filter(Q(orig_amount > 100000) | Q(dest_amount > 100000))`

I had to test this to be certain, but apparently Q objects don't support additional calls to `filter`:

```
>>> q = Q(saas.models.Transaction.objects.using('cowork').filter(orig_organization=cowork) | saas.models.Transaction.objects.using('cowork').filter(dest_organization=cowork))
>>> q.filter(orig_amount > 1000000)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'Q' object has no attribute 'filter'
```

And this makes sense to me now, because there's no obvious way to combine new criteria into a complex expression.

Maybe this is narrow-minded - just because Django doesn't offer it doesn't mean it's a good API. Or maybe it's good sense, because no Django developer would think to look for something that isn't Django-esque. What do you think?